node/full.go

```diff
@@ -217,6 +222,27 @@ func (n *FullNode) headerPublishLoop(ctx context.Context) {
 	}
 }
 
+func (n *FullNode) fraudProofPublishLoop(ctx context.Context) {
+	for {
+		select {
+		case fraudProof := <-n.blockManager.GetFraudProofOutChan():
+			n.Logger.Info("generated fraud proof: ", fraudProof.String())
+			fraudProofBytes, err := fraudProof.Marshal()
+			if err != nil {
+				n.Logger.Error("failed to serialize fraud proof", "error", err)
```

Comment:
I know this shouldn't happen, but in case of an error, we shouldn't gossip.

Reply:
fixed